### PR TITLE
enchilada: Add OnePlus 6/6T installer config

### PR DESCRIPTION
**Pull request by SevralT**
Hello! This is universal config for OnePlus 6/6T. Ubuntu Touch successfully installed on OnePlus 6/6T with this config.

### CODE REVIEW

**Comment by SevralT** [comment]
@Flohack74 And can you host boot and vbmeta on cdimage server??

File: v2/devices/enchilada.yml [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    image: "phone_power_down"
+    button: true
+  recovery:
+    title: "Reboot to Recovery"
+    description: "With the device still at the 'Fastboot Mode' screen (if not, power off the device and press and hold volume down and power buttons until the screen turns on), select the 'Recovery Mode' option by using the volume buttons and press the power button to enter."
+    image: "phone_power_down"
+    button: true
+  downgrade_android:
+    title: "Downgrade Android"
+    description: "If the device is running an Android version above 9.0, you have to downgrade to the last 9.0 release. Please make sure to flash OxygenOS 9 before proceeding."
```
Comment (Intrinsically-Sublime) [inline v2/devices/enchilada.yml:19]: description: "For ensured compatibility you should install Lineage 16.0 to the desired slot before flashing Ubuntu Touch. Other Android 9.0 based OS's may be compatible but not 10.0 or above"

**Review by Intrinsically-Sublime — CHANGES_REQUESTED** [review]
I think it should say Lineage 16.0 not Android 9.0. I bricked my phones twenty times trying over Oxygen 9.0.16 and 9.0.17. I tried every combination of formatting and wiping data. I even hard bricked it twice requiring the MSM tool to fix it.  It worked correctly the first time after installing Lineage 16.0. I did not install gapps but I did install LOS16.0 to both slots before installing Ubuntu Touch. Apparently it does something to the file system structure that is required.

**Comment by SevralT** [comment]
> I think it should say Lineage 16.0 not Android 9.0. I bricked my phones twenty times trying over Oxygen 9.0.16 and 9.0.17. I tried every combination of formatting and wiping data. I even hard bricked it twice requiring the MSM tool to fix it.  It worked correctly the first time after installing Lineage 16.0. I did not install gapps but I did install LOS16.0 to both slots before installing Ubuntu Touch. Apparently it does something to the file system structure that is required.

Hello! At all, do you know, why we using OxygenOS 9?? We needn't LineageOS. We are using OxygenOS 9 as a base, because 10 and 11 have other bootloader. Kernel is halium-9.0 based. You can use any OxygenOS 9.x.x.

**Comment by Intrinsically-Sublime** [comment]
I am not sure why but even after using the msm tool to install Oxygen 9.0.16 it would still brick to fastboot only once Ubuntu was finished installing. If I then tried switching to a different slot it would brick to Qualcomm crash dump loop where I had to remove the battery because it would reboot itself and go directly to crash dump. The only way to recover at that point was to plug it into a computer while holding vol+/vol- without the battery installed to get to EDL mode and reflash with the MSM tool. 

This thread says we must use Lineage https://forums.ubports.com/topic/5831/halium-9-ci-ota-build/10
Also the recovery tool and loading animation are Lineage based which led me to try Lineage.

My phone had Postmarket OS on it prior to flashing Oxygen 9.0.17 via fastboot to both slots and then turning on developer mode  in Lineage before trying to flash Ubuntu touch using the flasher and this config file. I do not think this matters seeing as I formatted the partitions multiple times using TWRP during different tries.

From further research it looks like OOS 9.0.17 is special in that it prepare the file system for OOS 10. But that does not explain why after flashing with the MSM tool to 9.0.16 it would still brick the phone.

**Comment by SevralT** [comment]
> I am not sure why but even after using the msm tool to install Oxygen 9.0.16 it would still brick to fastboot only once Ubuntu was finished installing. If I then tried switching to a different slot it would brick to Qualcomm crash dump loop where I had to remove the battery because it would reboot itself and go directly to crash dump. The only way to recover at that point was to plug it into a computer while holding vol+/vol- without the battery installed to get to EDL mode and reflash with the MSM tool. 
> 
> 
> 
> This thread says we must use Lineage https://forums.ubports.com/topic/5831/halium-9-ci-ota-build/10
> 
> Also the recovery tool and loading animation are Lineage based which led me to try Lineage.
> 
> 
> 
> My phone had Postmarket OS on it prior to flashing Oxygen 9.0.17 via fastboot to both slots and then turning on developer mode  in Lineage before trying to flash Ubuntu touch using the flasher and this config file. I do not think this matters seeing as I formatted the partitions multiple times using TWRP during different tries.
> 
> 
> 
> From further research it looks like OOS 9.0.17 is special in that it prepare the file system for OOS 10. But that does not explain why after flashing with the MSM tool to 9.0.16 it would still brick the phone.

Again!!! We needn't LineageOS. Recover is based on LineageOS but lol, we needn't LineageOS. Old images have no booted because we used old halium-generic_arm64. I had this problem with rebooting to fastboot too. It was fixed by adding vbmeta with disable verity and verification. It tested by few people. Try to flash OOS via TWRP.

**Comment by SevralT** [comment]
> I am not sure why but even after using the msm tool to install Oxygen 9.0.16 it would still brick to fastboot only once Ubuntu was finished installing. If I then tried switching to a different slot it would brick to Qualcomm crash dump loop where I had to remove the battery because it would reboot itself and go directly to crash dump. The only way to recover at that point was to plug it into a computer while holding vol+/vol- without the battery installed to get to EDL mode and reflash with the MSM tool.
> 
> This thread says we must use Lineage https://forums.ubports.com/topic/5831/halium-9-ci-ota-build/10
> Also the recovery tool and loading animation are Lineage based which led me to try Lineage.
> 
> My phone had Postmarket OS on it prior to flashing Oxygen 9.0.17 via fastboot to both slots and then turning on developer mode in Lineage before trying to flash Ubuntu touch using the flasher and this config file. I do not think this matters seeing as I formatted the partitions multiple times using TWRP during different tries.
> 
> From further research it looks like OOS 9.0.17 is special in that it prepare the file system for OOS 10. But that does not explain why after flashing with the MSM tool to 9.0.16 it would still brick the phone.

Few minutes ago, I successfully installed Ubuntu Touch on OnePlus 6T.

**Comment by ivoxavier** [comment]
Three days ago I've successfully flashed using Stock Android 9 into UT Release Candidate. Yesterday an OTA was also applied normally. 

So far, no major problems. Everything mentioned on the devices page about this accurate.

Just having issues with the uNav app and I can't detect my 6T by adb on terminal, even though developer mode is enabled. 

Calls, sms work as expected.

**Comment by Intrinsically-Sublime** [comment]
@SevralT Using the OFFICIAL way of flashing OOS9 to my phone (MSM tool) followed by Ubuntu touch repeatedly caused my phone to brick. Flashing both slots, both recoveries and the persistent via fastboot also did not work. The only way I was able to successfully install Ubuntu was after Lineage. Maybe you had already had Lineage on your phone prior to OOS9 or maybe flashing OOS9 with TWRP did something to the partition table that the OFFICIAL way does not do. My phone never had any custom Android ROMs on it, just PostmarketOS and Ubuntu.

The ONLY way I could install Ubuntu was AFTER installing Lineage. Luckily I am patient/determined enough to get it installed even if it did take hours of repeated attempts and repeatedly bricking the phone over two days. It was such a pain in the ass I had to find a Windows computer and fight with it to get the MSM tool to install OOS9 since the only Windows computer I had access to was Win10 with USB3 ports that required a hub to downgrade the USB driver mode to USB2.

Maybe find an untouched 6T and try installing over an OFFICIAL OOS9 install (not a repackaged version installed via TWRP). Or maybe try flashing your phone with the OFFICIAL OOS9 via the OFFICIAL msm tool and see if you can install over the stock OOS9 installed via the OFFICIAL way. Maybe you can explain why it was IMPOSSIBLE to install over an OFFICIAL install of the recommended OS? Either way the instructions seem wrong as simply installing over Android 9 did NOT work.  It seems easier to have people install Lineage before trying to install Ubuntu touch rather than dealing with lots of bricked devices. Or at least tell them to install OOS9 via TWRP not the official way.

I am glad your installs went well but not everyone's will. Recommending a way that prevents issues makes sense to me over sticking with a method that has proven to cause issues for at least one person coming from a completely stock OOS9 install. If it requires flashing OOS9 via TWRP than that is what the instructions must say. NOT just Android 9. Leave it, change it, in the end it does not matter to me as I figured out how to install it and I will not be the one helping people figure out why their phone is bricked and how to fix it.

**Comment by Flohack74** [comment]
My experience with Oneplus 5T was that the Oxygen vendor partition was not compatible with our builds from Lineage base. I think that could contribute. IDK what Oneplus is doing but the idea of an "independent" vendor partition seem to be fake news ;)

**Comment by SevralT** [comment]
> My experience with Oneplus 5T was that the Oxygen vendor partition was not compatible with our builds from Lineage base. I think that could contribute. IDK what Oneplus is doing but the idea of an "independent" vendor partition seem to be fake news ;)
> 
> 

LineageOS have no vendor partition. (It is not building. We are using OOS vendor.). I installed UT successfully with OOS only ~10 times.

File: v2/devices/enchilada.yml [hunk 2]
```diff
@@ -0,0 +1,121 @@
+name: "OnePlus 6/6T"
+codename: "enchilada"
+formfactor: "phone"
+aliases: []
+doppelgangers: []
```
Comment (NotKit) [inline v2/devices/enchilada.yml:5]: Do fajita need a different config/boot image?

**Comment by NotKit** [comment]
The config looks sane to me. Regarding LineageOS discussion, the port is supposed to use stock OOS 9 vendor, so installing LineageOS 16 is not required. The problem encountered by @Intrinsically-Sublime was likely due to vbmeta not getting flashed, which is addressed in current version.

**Review by NotKit — APPROVED** [review]

**Comment by NotKit** [comment]
Confirmed installation on OnePlus 6/enchilada

File: v2/devices/enchilada.yml (hunk 2 above)
Comment (SevralT) [reply v2/devices/enchilada.yml:5]: All is same.

**Pull request merged**